### DISCUSSION TITLE
Github.dev web editor live preview for html/js apps

### DISCUSSION THREAD

**Discussion by mmiscool** [community/community#15438] [Codespaces]
Hello,
I have a relatively small simple project composed of vanilla html, js & css.
https://github.dev/mmiscool/serialTerminal.com
If I open the project in using the "github.dev" browser based VS code I can edit the project without issue. I have how ever not been able to figure out a method to show a live preview. It was my hope that VS Code in the browser would at least match the capabilities of https://jsfiddle.net/ .

Is there any method to get a real time capability to run the code in the browser? Perhaps a VS Code plugin?

**Comment by soutiagojose** [comment]
This site generates a preview of the page's html code
https://htmlpreview.github.io/

**Reply by mmiscool** [reply]
The trouble with that approach is you would need to commit and push each time you wanted to use the preview.

**Comment by jfullstackdev** [comment]
there are so many ways to achieve it, 

but take note `github.dev` is not the the same as Codespaces, 

https://docs.github.com/en/codespaces/the-githubdev-web-based-editor#codespaces-and-the-web-based-editor

maybe these tutorials will help 

https://www.youtube.com/watch?v=vDXPkAdqSyI

https://www.youtube.com/watch?v=8KwoKgYz85k

**Reply by mmiscool** [reply]
Yes. Specifically I am trying to be able to develop web applications with out needing the virtual machine backing of a codespaces instance. Codespaces cost money and I understand the difrence between codespaces and github.dev

I was just hoping that one of the simplest usecases covered by sites like jsfiddle and codepen might be possible to replicate using a more complete VScode implementation in the browser.

**Reply by jfullstackdev** [reply]
> Codespaces cost money and I understand the difrence between codespaces and github.dev

![image](https://user-images.githubusercontent.com/47092464/182010483-749fcfc9-8323-4cee-bd76-d5b7f5e32344.png)

**Comment by Lexyth** [comment]
Does someone happen to have an answer to the actual question? I would be very interested in that, too.

**Reply by T3sT3ro** [reply]
For now, [Codeswing](https://marketplace.visualstudio.com/items?itemName=codespaces-Contrib.codeswing) works for now, but is not perfect (for example initializing new swing in a directory overwrites the content of files). I've only started using it, but it allows for previewing the app in the same tab. The setup is not ideal though.

I've tried the webcontainers extensions, but had no luck with them. Sadly, neither [github.dev](https://github.dev/) nor [vscode.dev](https://vscode.dev/) use webcontainers to allow for usage of node in the browser (like [stackblitz.com](https://stackblitz.com/) does)

**Reply by whimsyniche** [reply]
That's the one! Codeswing

**Comment by whimsyniche** [comment]
I think @planetoftheweb talked about this in on of this videos. I forget where to find this or how to do this.

**Reply by whimsyniche** [reply]
See that someone mentioned CodeSwing. I just installed in my GH web editor.

**Comment by omundy** [comment]
I found that you can preview your work using Codespaces by either [running a simple http server](https://stackoverflow.com/a/75497415/441878) or by installing a [Live Server](https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer) extension.

**Comment by Donglan-digital** [comment]
As a beginner, I came across this question as well...got here to find out any solution...

**Comment by Nikhil-Info** [comment]
Hey! 👋

You're right - as of now, [**[github.dev](https://github.dev/)**](https://github.dev) (the browser-based VS Code) doesn't support live preview or running HTML/JS/CSS in real time like **JSFiddle** or **CodePen**. It's mainly a lightweight code editor without execution or server capabilities.

If you're looking for a live preview experience:

* You might want to try using **GitHub Codespaces**, which does support live preview (via a dev container and server).
* Alternatively, tools like **StackBlitz**, **CodeSandbox**, or even a local VS Code with the **Live Server** extension can give you that instant feedback.

Would love to see GitHub.dev support this natively someday - it’d be super useful for frontend projects like yours!
